### PR TITLE
fix

### PR DESCRIPTION
**Pull request by andyFang94**

### CODE REVIEW

**Comment by VictorHAS** [comment]
Heyya, thanks for checking this, didn't realize, but I think the change shouldn't be there, since receivedTopic is unique.

I think the right thing would be:
`if ([topic].flat().some(rTopic => matches(rTopic, receivedTopic))) {`

matches from mqtt-pattern:

`matches(pattern : String, topic : String) : Boolean`
Validates whether topic fits the pattern. Ignore parameters.

could make this change and commit again? then i accept the PR

File: lib/useSubscription.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function useSubscription(
 
   const callback = useCallback(
     (receivedTopic: string, receivedMessage: any) => {
-      if ([topic].flat().some(rTopic => matches(topic, rTopic))) {
+      if ([receivedTopic].flat().some(rTopic => matches(topic, rTopic))) {
```
Comment (VictorHAS) [inline lib/useSubscription.tsx:25]: I think the right thing would be:
`if ([topic].flat().some(rTopic => matches(rTopic, receivedTopic))) {`

**Review by VictorHAS — CHANGES_REQUESTED** [review]
Heyya, thanks for checking this, didn't realize, but I think the change shouldn't be there, since receivedTopic is unique.

I think the right thing would be:
`if ([topic].flat().some(rTopic => matches(rTopic, receivedTopic))) {`

matches from mqtt-pattern:

`matches(pattern : String, topic : String) : Boolean`
Validates whether topic fits the pattern. Ignore parameters.

could make this change and commit again? then i accept the PR

File: lib/useSubscription.tsx (hunk 1 above)
Comment (andyFang94) [reply lib/useSubscription.tsx:25]: I agree with you，I made a mistake。

**Comment by andyFang94** [comment]
> Heyya, thanks for checking this, didn't realize, but I think the change shouldn't be there, since receivedTopic is unique.
> 
> I think the right thing would be:
> `if ([topic].flat().some(rTopic => matches(rTopic, receivedTopic))) {`
> 
> matches from mqtt-pattern:
> 
> `matches(pattern : String, topic : String) : Boolean`
> Validates whether topic fits the pattern. Ignore parameters.
> 
> could make this change and commit again? then i accept the PR

**Review by VictorHAS — APPROVED** [review]

**Pull request merged**